src/headless-cms/components/cmsSidebarContent/CmsSidebarContent.tsx

```diff
@@ -1,32 +1,95 @@
+import useLocale from '../../../hooks/useLocale';
+import { Language } from '../../../types';
+import styles from './cmsSidebarContent.module.scss';
+import CmsSidebarContentCard from './CmsSidebarContentCard';
 import CmsSidebarContentLayoutLinkList from './CmsSidebarContentLayoutLinkList';
 
 type Props = {
   content?: PageFieldsFragment['sidebar'];
 };
 
 const CmsSidebarContent: React.FC<Props> = ({ content }) => {
+  const locale = useLocale();
+
   return (
-    <>
-      {content?.map((item) => {
+    <ul className={styles.container}>
+      {content?.map((item, i) => {
         if (item?.__typename === 'LayoutLinkList') {
           return (
-            <CmsSidebarContentLayoutLinkList
-              key={item.title}
-              title={item.title}
-              links={item.links}
-              description={item.description}
-              anchor={item.anchor}
-            />
+            <li key={item.title}>
+              <CmsSidebarContentLayoutLinkList
+                title={item.title}
+                links={item.links}
+                description={item.description}
+                anchor={item.anchor}
+              />
+            </li>
           );
         }
 
+        if (item?.__typename === 'LayoutPages') {
+          return item?.pages?.map((page) => {
+            const title = page?.title;
+            const uri = page?.uri;
+
+            if (!title || !uri) {
+              return null;
+            }
+
+            return (
+              <li key={page?.id}>
+                <CmsSidebarContentCard
+                  title={title}
+                  url={createCmsLink(uri, locale)}
+                  image={page?.featuredImage?.node?.mediaItemUrl || undefined}
+                  imageAlt={page?.featuredImage?.node?.altText || undefined}
+                />
+              </li>
+            );
+          });
+        }
+
+        if (item?.__typename === 'LayoutArticles') {
+          return item?.articles?.map((article) => {
+            const title = article?.title;
+            const uri = article?.uri;
+
+            if (!title || !uri) {
+              return null;
+            }
+
+            return (
+              <li key={article?.id}>
+                <CmsSidebarContentCard
+                  title={title}
+                  url={createCmsLink(uri, locale)}
+                  image={
+                    article?.featuredImage?.node?.mediaItemUrl || undefined
+                  }
+                  imageAlt={article?.featuredImage?.node?.altText || undefined}
+                />
+              </li>
+            );
+          });
+        }
+
         return null;
       })}
-    </>
+    </ul>
   );
 };
 
 export default CmsSidebarContent;
+
+function createCmsLink(cmsUri: string, language: Language) {
+  const baseUri = `/cms-page${cmsUri}`;
+
+  if (language === 'fi') {
+    return baseUri;
+  }
+
+  return `/${language}${baseUri}`;
+}
```

Comment:
This implementation has problem when linked to pages with other language, for example the following url is generated when linking to english page:

`/en/cms-page/en/learning-materials/elementary-school`

There is also `getCmsPath` util in this project tat tries to accomplish the same task

```
const getCmsPath = (slug?: string | null): string => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
``` 

Maybe `getCmsPath` could be refactored little bit and used also here. If `getCmsPath` would receive locale as second parameter some other places need refactoring too, but I think it is the better solution.

Reply:
I checked out your suggestion, but understood while doing it that this old utility was alone as is when combined with `stripLocaleFromUri` from `headless-cms` utils.

The url gets passed on to a `next/link` component that adds the correct locale into the link.